src/libraries/Microsoft.Extensions.FileProviders.Physical/src/Internal/FileSystemInfoHelper.cs

```diff
@@ -27,5 +28,40 @@ public static bool IsExcluded(FileSystemInfo fileSystemInfo, ExclusionFilters fi
 
             return false;
         }
+
+        public static DateTime? GetFileLinkTargetLastWriteTimeUtc(string filePath)
+        {
+#if NETCOREAPP
+            var fileInfo = new FileInfo(filePath);
+            if (fileInfo.Exists)
+            {
+                return GetFileLinkTargetLastWriteTimeUtc(fileInfo);
+            }
+#endif
+            return null;
+        }
+
+        // If file is a link and link target exists, return target's LastWriteTimeUtc.
+        // If file is a link, and link target does not exists, return DateTime.MinValue
+        //   since the link's LastWriteTimeUtc doesn't convey anything for this scenario.
+        // If file is not a link, return null to inform the caller that file is not a link.
+        public static DateTime? GetFileLinkTargetLastWriteTimeUtc(FileInfo fileInfo)
+        {
+#if NETCOREAPP
+            Debug.Assert(fileInfo.Exists);
+            if (fileInfo.LinkTarget != null)
+            {
+                FileSystemInfo targetInfo = fileInfo.ResolveLinkTarget(returnFinalTarget: true);
```

Comment:
@Jozkee  have you considered keeping all these libs .NET Standard 2.0 and referencing the code that defines `ResolveLinkTarget` (and other things if needed)? We are already doing something like this for `Microsoft.IO.Redist` which targets `net471`. I am just wondering how much effort it would require to keep it NS2.0.

@jeffhandley @ericstj how long do we plan to keep targeting `netstandard2.0` and `net461` for the `Microsoft.Extensions*` libs?

Reply:
> @jeffhandley Jeff Handley FTE @ericstj Eric St. John FTE how long do we plan to keep targeting netstandard2.0 and net461 for the Microsoft.Extensions* libs?

At the moment we have no plans for dropping `netstandard2.0` (and therefor `net461`) from the extensions.  Perhaps it's something that could be considered in the future if this limitation hinders development considerably.  They were intentionally omitted from https://github.com/aspnet/Announcements/issues/324.  cc @davidfowl @DamianEdwards @maryamariyan

Reply:
Yes, we're keeping ns2.0 and net461 until further notice.

Reply:
We can do a different (and more efficient) approach to get the target's Last write time, see @ericstj's suggestion https://github.com/dotnet/runtime/pull/55664#discussion_r669872218.
But (not) saving a couple of allocations is not a big issue considering that the performance penalty is relatively low compared to other pieces of the code that allocate much more.

Reply:
> We can do a different (and more efficient) approach to get the target's Last write time

@Jozkee I wanted to know whether it would be possible to make this fix work for all TFMs (6 + NS2 + net461), because currently, it's going to work only for .NET 6.

Reply:
It is, but we can defer fixing this on other TFMs if no one is asking for it. 
Do you think we can add a dependency to MS.IO.Redist here to utilize the same set of Symbolic Link APIs in order to fix this in ns2.0? 
cc @jeffhandley.

Reply:
I think to fix on older TFMs it needs more than just MS.IO.Redist.  I believe the implementation of the IO API you are using here (or would add) would depend on the system-native PALs which we don't consider part of the public API.  Having a package depend on those is fragile at the least, and might not even work if the versions of those don't have the exports you need on older frameworks.  Adding private copies of the system-native PALs (similar to what System.IO.Ports does) is expensive too.  Something to consider around OOB'ing our IO API.

Reply:
Ahh, good info @ericstj. So far, we don't have any requests to have this fix apply to previous versions; I'm OK with this being net6.0+ until such requests come in.